### PR TITLE
[core] Introduce rollback to snapshot

### PR DESCRIPTION
**Pull request by JingsongLi**
### Purpose

Rollback table's state to a specific snapshot.

### Tests

`FileStoreTableTestBase.testRollbackTo`

### API and Format

Yes Table.rollbackTo

### Documentation

Yes

### CODE REVIEW

File: docs/content/maintenance/manage-snapshots.md [hunk 1]
```diff
@@ -24,7 +24,11 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-# Expiring Snapshots
+# Manage Snapshots
+
+This section will describe the management and behavior related to snapshots.
+
+## Expiring Snapshots
```
Comment (yuzelin) [inline docs/content/maintenance/manage-snapshots.md:31]: Manage, Expire, Rollback (all not using -ing form)?

File: docs/content/maintenance/manage-snapshots.md [hunk 2]
```diff
@@ -73,4 +77,57 @@ Please note that too short retain time or too small retain number may result in:
...
+Run the following command:
+
+```bash
+<FLINK_HOME>/bin/flink run \
+    /path/to/paimon-flink-action-{{< version >}}.jar \
+    rollback-to \
+    --warehouse <warehouse-path> \
+    --database <database-name> \ 
+    --table <table-name> \
+    [--snapshot <snapshot-id>]
```
Comment (yuzelin) [inline docs/content/maintenance/manage-snapshots.md:101]: `snapshot` is not optional.

**Comment by JingsongLi** [comment]
Thanks @yuzelin for the review, addressed

**Review by yuzelin — APPROVED** [review]
LGTM.

**Pull request merged**